doc/cygnus-ngsi/flume_extensions_catalogue/ngsi_cartodb_sink.md

```diff
@@ -0,0 +1,455 @@
+#<a name="top"></a>NGSICartoDBSink
+Content:
+
+* [Functionality](#section1)
+    * [Mapping NGSI events to flume events](#section1.1)
+    * [Mapping Flume events to CartoDB data structures](#section1.2)
+        * [PostgreSQL databases and schemas naming conventions](#section1.2.1)
+        * [PostgreSQL tables naming conventions](#section1.2.2)
+        * [Raw-based storing](#section1.2.3)
+        * [Distance-based storing (beta)](#section1.2.4)
+    * [Example](#section1.3)
+        * [Flume event](#section1.3.1)
+        * [Table names](#section1.3.2)
+        * [Raw-based storing](#section1.3.3)
+        * [Distance-based storing](#section1.3.4)
+* [Administration guide](#section2)
+    * [Configuration](#section2.1)
+    * [Use cases](#section2.2)
+    * [Important notes](#section2.3)
+        * [`NGSICartoDBSink` and non-geolocated entities](#section2.3.1)
+        * [Batching](#section2.3.2)
+* [Programmers guide](#section3)
+    * [`NGSICartoDBSSink` class](#section3.1)
+    * [Authentication and authorization](#section3.2)
+
+##<a name="section2"></a>Functionality
+`com.iot.telefonica.cygnus.sinks.NGSICartoDBSink`, or simply `NGSICartoDBSSink` is a cygnus-ngsi sink designed to persist NGSI-like context data events within [CartoDB](https://cartodb.com/). Usually, such a context data is notified by a [Orion Context Broker](https://github.com/telefonicaid/fiware-orion) instance, but could be any other system speaking the <i>NGSI language</i>.
+
+Independently of the data generator, NGSI context data is always transformed into internal Flume events at cygnus-ngsi sources. In the end, the information within these Flume events must be mapped into specific CartoDB data structures at the Cygnus sinks.
+
+Next sections will explain this in detail.
+
+[Top](#top)
+
+###<a name="section1.1"></a>Mapping NGSI events to flume events
+Notified NGSI events (containing context data) are transformed into Flume events (such an event is a mix of certain headers and a byte-based body), independently of the NGSI data generator or the final backend where it is persisted.
+
+This is done at the cygnus-ngsi Http listeners (in Flume jergon, sources) thanks to [`NGSIRestHandler`](./orion_rest_handler.md). Once translated, the data (now, as a Flume event) is put into the internal channels for future consumption (see next section).
+
+[Top](#top)
+
+###<a name="section1.2"></a>Mapping Flume events to CartoDB data structures
+CartoDB is based on [PostgreSQL](http://www.postgresql.org/) and [PostGIS](http://postgis.net/) extensions. It organizes the data in databases (one per organization), schemas (una per project within an organization) and tables (a schema may have one or more tables). Such organization is exploited by `NGSICartoDBSink` each time a Flume event is going to be persisted.
+
```

Comment:
It depends on what we consider a "project" and a "user". Let's discuss it offline ;)